cliv2/internal/proxy/proxy.go

```diff
@@ -146,6 +146,7 @@ func (p *WrapperProxy) ProxyInfo() *ProxyInfo {
 func (p *WrapperProxy) replaceVersionHandler(r *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
 	err := p.addHeaderFunc(r)
 	if err != nil {
```

Comment:
Once https://github.com/snyk/go-application-framework/pull/213 lands,

```suggestion
	if errors.Is(err, middleware.ErrAuthenticationFailed) {
```

would be more accurate.